common/gateway-storage/migrations/20240910120000_generic_client_id.sql

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+CREATE TABLE clients (
+   id              INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+   client_type     TEXT NOT NULL CHECK(client_type IN ('entry mixnet', 'entry wireguard', 'exit wireguard'))
+);
+
+INSERT INTO clients (id, client_type)
+SELECT id, 'entry mixnet'
+FROM shared_keys;
+
+CREATE TABLE shared_keys_tmp (
+   client_id                                INTEGER NOT NULL PRIMARY KEY REFERENCES clients(id),
+   client_address_bs58                      TEXT NOT NULL UNIQUE,
+   derived_aes128_ctr_blake3_hmac_keys_bs58 TEXT NOT NULL
+);
+
+INSERT INTO shared_keys_tmp (client_id, client_address_bs58, derived_aes128_ctr_blake3_hmac_keys_bs58)
+SELECT id as client_id, client_address_bs58, derived_aes128_ctr_blake3_hmac_keys_bs58 FROM shared_keys;
+
+CREATE TABLE available_bandwidth_tmp (
+   client_id INTEGER NOT NULL PRIMARY KEY REFERENCES clients(id),
+   available INTEGER NOT NULL,
+   expiration TIMESTAMP WITHOUT TIME ZONE
+);
+
+INSERT INTO available_bandwidth_tmp
+SELECT * FROM available_bandwidth;
+
+DROP TABLE available_bandwidth;
+ALTER TABLE available_bandwidth_tmp RENAME TO available_bandwidth;
+
+CREATE TABLE received_ticket_tmp (
```

Comment:
why are you recreating this table if nothing has changed in it? (the same applies to other tables too)

Reply:
Because `received_ticket`'s `ticket_id` references `received_ticket`'s `id`. All the references cascade because of one change. Things would have been much easier if the `ALTER TABLE MODIFY COLUMN` would have worked in sqlite, but since we have to do the create and rename "hack", this affects inter-table dependencies.